examples/platform/nxp/se05x/DeviceAttestationSe05xCredsExample_v2.cpp

```diff
@@ -0,0 +1,301 @@
+        /* Set time stamp length */
+        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(CD_TIME_STAMP_LEN_SE05X_ID, &tmp, 1), CHIP_ERROR_INTERNAL);
+        if (time_stamp.size() > 0)
+        {
+            /* Set time stamp data */
+            VerifyOrReturnError(CHIP_NO_ERROR ==
+                                    se05xSetCertificate(CD_TIME_STAMP_DATA_SE05X_ID, time_stamp.data(), time_stamp.size()),
+                                CHIP_ERROR_INTERNAL);
+        }
+
+        if (message_to_sign.size() >= 16)
+        {
+            /* Set attestation challenge */
+            VerifyOrReturnError(CHIP_NO_ERROR ==
+                                    se05xSetCertificate(CD_ATTEST_CHALLENGE_SE05X_ID, (message_to_sign.end() - 16), 16),
+                                CHIP_ERROR_INTERNAL);
+        }
+    }
+    else
+    {
+        tagValue = msg_reader.GetTag();
+        VerifyOrReturnError(TLV::ContextTag(1) == tagValue, CHIP_ERROR_INVALID_TLV_TAG);
+        /* Get nocsr */
+        ByteSpan csr_data;
+        msg_reader.Get(csr_data);
+        uint8_t tmp = csr_data.size();
+        /* Set nocsr length */
+        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(NOCSR_CSR_LEN_SE05X_ID, &tmp, 1), CHIP_ERROR_INTERNAL);
+        /* Set nocsr data */
+        se05x_delete_key(NOCSR_CSR_DATA_SE05X_ID);
+        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(NOCSR_CSR_DATA_SE05X_ID, csr_data.data(), csr_data.size()),
+                            CHIP_ERROR_INTERNAL);
+
+        msg_reader.Next();
+        tagValue = msg_reader.GetTag();
+        VerifyOrReturnError(TLV::ContextTag(2) == tagValue, CHIP_ERROR_INVALID_TLV_TAG);
+        /* Get nocsr nonce */
+        ByteSpan nocsr_nonce;
+        msg_reader.Get(nocsr_nonce);
+        /* Set nocsr nonce data */
+        VerifyOrReturnError(CHIP_NO_ERROR ==
+                                se05xSetCertificate(NOCSR_CSR_NONCE_DATA_SE05X_ID, nocsr_nonce.data(), nocsr_nonce.size()),
+                            CHIP_ERROR_INTERNAL);
+
+        if (message_to_sign.size() >= 16)
+        {
+            /* Set attestation challenge */
+            VerifyOrReturnError(CHIP_NO_ERROR ==
+                                    se05xSetCertificate(NOCSR_ATTEST_CHALLENGE_SE05X_ID, (message_to_sign.end() - 16), 16),
+                                CHIP_ERROR_INTERNAL);
+        }
+    }
+
+    uint8_t signature_se05x[Crypto::kMax_ECDSA_Signature_Length_Der] = { 0 };
+    size_t signature_se05x_len                                       = sizeof(signature_se05x);
+
+    if (sign_cert_decl_attest)
+    {
+        err = se05xPerformInternalSign(CD_DEV_ATTESTATION_KEY_SE05X_ID, signature_se05x, &signature_se05x_len);
+        se05x_delete_key(CD_ATTEST_NONCE_DATA_SE05X_ID);
+        se05x_delete_key(CD_TIME_STAMP_LEN_SE05X_ID);
+        se05x_delete_key(CD_TIME_STAMP_DATA_SE05X_ID);
+        se05x_delete_key(CD_ATTEST_CHALLENGE_SE05X_ID);
+        sign_cert_decl_attest = 0;
```

Comment:
This seems to be assuming that the calls to this function will come in a certain order in terms of whether we're signing a CSR or the attestation bits.  What guarantees this order?  Note that two CSRRequests can happen in a row....

Reply:
This limitation will be removed in the subsequent PR. (Put a note for the same). After updating our HSM side implementation, the signing will be independent of the calls. 
Also this is not the default se05x DAC provider class. The default (ExampleSe05xDACProvider) will sign the message without paring the TLV data.